src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -546,8 +563,7 @@ internal void WriteCentralDirectoryFileHeader()
             if (_cdUnknownExtraFields != null)
                 ZipGenericExtraField.WriteAllBlocks(_cdUnknownExtraFields, _archive.ArchiveStream);
 
-            if (_fileComment != null)
-                writer.Write(_fileComment);
+            writer.Write(_fileComment);
```

Comment:
for empty arrays we can save some time by avoiding the call to `writer.Write` (which calls `Stream.Write` which does some checks (CanWrite, IsClosed etc) before actually realizing that the buffer is empty and there is nothing to do)

```suggestion
            if (_fileComment.Length > 0)
                writer.Write(_fileComment);
```